### PR TITLE
Connect pano to GQL

### LINKED ISSUE #544: feat(gql): Create `createPost` mutation

### LINKED ISSUE #545: feat(gql): Create `updatePost` mutation

### LINKED ISSUE #582: feat(gql): create `removePost` mutation

### LINKED ISSUE #588: epic: gql & pasaport integration
### What is this epic about?

WIP, creating this so i can take notes, will update later to a proper epic.

Track progress at #572 

we should start developing our own "envelop" plugins, and this integration should be handled that way as well.

example custom auth plugin developed using "envelop": https://the-guild.dev/graphql/yoga-server/docs/features/envelop-plugins#example

### Tasks for this epic

- [x] #589
- [x] #590

### LINKED ISSUE #589: feat(gql): extend `KampusGQLContext` to include a `pasaport` field

### LINKED ISSUE #590: feat(gql): Add `Viewer` type and `Query.viewer` resolver

### PR DESCRIPTION
**Pull request by usirin**
This PR connects pano to GQL.

Closes #544 
Closes #545 
Closes #582 
Closes #588 
Closes #589 
Closes #590

### CODE REVIEW

File: apps/kampus/app/pano/features/post-list/PostItem.tsx [hunk 1]
```diff
@@ -1,24 +1,15 @@
+"use client";
+
 import NextLink from "next/link";
+import { graphql, useFragment } from "react-relay";
 
 import { cn } from "@kampus/ui-next/utils";
 
-import { MoreOptionsDropdown } from "./MoreOptions";
+import { TimeAgo } from "~/../../packages/ui";
```
Comment (Ketcap) [inline apps/kampus/app/pano/features/post-list/PostItem.tsx]: wierd flex but okay, isn't it "@kampus/ui" ?

File: apps/kampus/app/pano/features/post-list/PostItem.tsx [hunk 2]
```diff
@@ -1,24 +1,15 @@
...
-  owner: string;
-  title: string;
-  url: string;
-};
-
-type PostItemProps = {
-  post: Post;
-  showContent?: boolean;
-};
+// import { MoreOptionsDropdown } from "./MoreOptions";
```
Comment (Ketcap) [inline apps/kampus/app/pano/features/post-list/PostItem.tsx]: Just delete it if it's commented

File: apps/kampus/app/pano/features/post-list/usePanoPostFragment.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (Ketcap) [inline apps/kampus/app/pano/features/post-list/usePanoPostFragment.ts]: Empty file but okay :D

File: apps/gql/package.json [hunk 4]
```diff
@@ -29,6 +29,7 @@
   "dependencies": {
     "@graphql-tools/schema": "9.0.19",
     "@kampus/gql-utils": "*",
+    "@kampus/next-auth": "^0.0.0",
```
Comment (Ketcap) [inline apps/gql/package.json]:
```suggestion
    "@kampus/next-auth": "*",
```
any reason why gql requires auth ?

**Comment by usirin** [comment]
dude, it's WIP, chilllllllllllllll <3 @Ketcap

**Review by rasitds — APPROVED** [review]
:100:

File: packages/relay/environment.ts [hunk 5]
```diff
@@ -10,10 +10,11 @@ import {
   type Variables,
 } from "relay-runtime";
 
-const HTTP_ENDPOINT = "https://gql.dev.kamp.us/graphql";
 const IS_SERVER = typeof window === typeof undefined;
 const CACHE_TTL = 5 * 1000; // 5 seconds, to resolve preloaded results
 
+const HTTP_ENDPOINT = IS_SERVER ? (process.env.NEXT_PUBLIC_GQL_URL as string) : "/gql";
```
Comment (usirin) [inline packages/relay/environment.ts:16]: This line requires apps using this package to have a rewrite rule configured for `/gql` url to real graphql url (eg `https://gql.dev.kamp.us/graphql`)

**Comment by usirin** [comment]
Merging this to unblock stuff, we can open bug tickets if we see something broken.

**Comment by cansirin** [comment]
> Merging this to unblock stuff, we can open bug tickets if we see something broken.

Makes sense tbh! Sounds awesome, can't wait to develop all the features with GQL/relay 🤌🏻😍

**Pull request merged**